Source/Core/Core/PowerPC/GDBStub.cpp

```diff
@@ -320,25 +539,91 @@ static void WriteHostInfo()
           .c_str());
 }
 
+static void ProcessXFerCommand(const char* data, size_t paramsIndex)
+{
+  size_t offset = 0;
+  size_t length = 0;
+
+  while (s_cmd_bfr[paramsIndex] != ',')
+  {
+    offset <<= 4;
+    offset |= Hex2char(s_cmd_bfr[paramsIndex]);
+    paramsIndex++;
+  }
+
+  paramsIndex++;
+  while (paramsIndex < s_cmd_len)
+  {
+    length <<= 4;
+    length |= Hex2char(s_cmd_bfr[paramsIndex]);
+    paramsIndex++;
+  }
+  static u8 reply[GDB_BFR_MAX];
+  memset(reply, 0, GDB_BFR_MAX);
+  if (length + 1 > GDB_BFR_MAX)
+    length = GDB_BFR_MAX - 1;
+
+  if (strlen(data) < static_cast<size_t>(length) + offset)
+  {
+    length = strlen(data) - offset;
+    reply[0] = 'l';
+  }
+  else
+  {
+    reply[0] = 'm';
+  }
+  memcpy(&reply[1], &data[offset], length);
+  return SendReply(reinterpret_cast<const char*>(reply));
+}
+
+static bool IsQuery(const char* query)
+{
+  return !strncmp((const char*)(s_cmd_bfr), query, strlen(query));
+}
+
 static void HandleQuery()
 {
   DEBUG_LOG_FMT(GDB_STUB, "gdb: query '{}'", CommandBufferAsString());
 
-  if (!strncmp((const char*)(s_cmd_bfr), "qAttached", strlen("qAttached")))
+  if (IsQuery("qAttached"))
+  {
     return SendReply("1");
-  if (!strcmp((const char*)(s_cmd_bfr), "qC"))
+  }
+  else if (IsQuery("qC"))
+  {
```

Comment:
Note that the old behavior here was checking if the string exactly matched `qC`, not if it started with `qC`.  I don't know if this is a problem.